spec/models/search/bookmark_query_spec.rb

```diff
@@ -114,11 +114,46 @@ def find_parent_filter(query_list)
         include({ term: { complete: "true" } })
     end
 
-    it "allows you to filter by bookmarkable tags" do
-      q = BookmarkQuery.new(parent: tag)
+    context "when querying as a guest" do
+      it "allows filtering by unrestricted bookmarkable tags" do
+        q = BookmarkQuery.new(parent: tag)
+        parent = find_parent_filter(q.generated_query.dig(:query, :bool, :must))
+        expect(parent.dig(:has_parent, :query, :bool, :filter)).to \
+          include({ term: { public_filter_ids: tag.id } })
+      end
+    end
+
+    context "when querying as a admin" do
+      before do
+        User.current_user = create(:admin)
+      end
+
+      it "allows filtering by restricted bookmarkable tags" do
+        q = BookmarkQuery.new(parent: tag)
+        parent = find_parent_filter(q.generated_query.dig(:query, :bool, :must))
+        expect(parent.dig(:has_parent, :query, :bool, :filter)).to \
+          include({ term: { general_filter_ids: tag.id } })
+      end
+    end
+
+    it "allows you to filter by bookmarkable language" do
+      q = BookmarkQuery.new(language_id: "ig")
       parent = find_parent_filter(q.generated_query.dig(:query, :bool, :must))
       expect(parent.dig(:has_parent, :query, :bool, :filter)).to \
-        include({ term: { filter_ids: tag.id } })
+        include({ term: { "language_id.keyword": "ig" } })
+    end
```

Comment:
Duplicated from the spec below "when querying as a registered user" > "allows filtering by restricted bookmarkable tags"